File: lib/phone_photo.dart
import 'package:img_info_picker/utils/date_time.util.dart';

import 'model/file_type.enum.dart';

class PhonePhoto {
  final String id;
  final String albumName;
  final String uri;
  final DateTime? lastModifiedDate;
  final DateTime? createdDate;
  final FileType? fileType;
  PhonePhoto(this.id, this.albumName, this.uri, this.lastModifiedDate, this.createdDate, this.fileType);

  PhonePhoto.fromMap(Map<String, dynamic> map):
    id = map["Id"],
    albumName = map["AlbumName"],
    uri = map["PhotoUri"],
    lastModifiedDate = DateTimeUtility.getDateFromStringMillisencode(map["LastModifiedDate"]),
    createdDate = DateTimeUtility.getDateFromStringMillisencode(map["CreatedDate"]),
    fileType = FileType.values[int.tryParse(map["FileType"]) ?? (FileType.values.length -1)];

  @override
  String toString() {
    return 'PhonePhoto{id: $id, albumName: $albumName, photoUri: $uri, lastModifiedDate: $lastModifiedDate, createdDate: $createdDate, fileType: $fileType}';
  }
}

File: android/src/main/kotlin/com/example/img_info_picker/PhonePhoto.kt
package com.tahamalas.custom_image_picker

class PhonePhoto(private val id: String, private val photoUri: String, private val albumName: String, private val videoTumbUrl :String = "", private val fileType: String= "", private val lastModifiedDate: String= "", private val createdDate: String= "") {
    fun toJson() : String{
        return "{" +
                "\"Id\": \"$id\", " +
                "\"AlbumName\": \"$albumName\", " +
                "\"PhotoUri\": \"$photoUri\"," +
                "\"LastModifiedDate\": \"$lastModifiedDate\"," +
                "\"CreatedDate\": \"$createdDate\"," +
                "\"FileType\": \"$fileType\"," +
                "\"VideoTumblr\": \"$videoTumbUrl\"" +
        "}"
    }
}